src/components/dashboard/RecoveryHeader/index.test.tsx

```diff
@@ -46,22 +41,80 @@ describe('RecoveryHeader', () => {
 
     const { queryByText } = render(
       <RecoveryContext.Provider value={{ state: [[{ queue }]] } as any}>
-        <_RecoveryHeader isOwner={false} isRecoverer queue={queue} supportsRecovery />
+        <_RecoveryHeader isProposalInProgress={false} isRecoverer queue={queue} supportsRecovery />
       </RecoveryContext.Provider>,
     )
 
     expect(queryByText('Recover this Account')).toBeTruthy()
   })
 
-  it('should not render the proposal widget when there is no queue for owners', () => {
+  it('should not render the proposal widget when there is no queue for recoverers and proposal is in progress', () => {
     const queue = [] as any
 
     const { container } = render(
       <RecoveryContext.Provider value={{ state: [[{ queue }]] } as any}>
-        <_RecoveryHeader isOwner isRecoverer={false} queue={queue} supportsRecovery />
+        <_RecoveryHeader isProposalInProgress isRecoverer queue={queue} supportsRecovery />
       </RecoveryContext.Provider>,
     )
 
     expect(container).toBeEmptyDOMElement()
   })
 })
+
+describe('useIsProposalInProgress', () => {
+  ;[
+    RecoveryEvent.EXECUTING,
+    RecoveryEvent.PROCESSING,
+    RecoveryEvent.REVERTED,
+    RecoveryEvent.PROCESSED,
+    RecoveryEvent.FAILED,
+  ].forEach((event) => {
+    it('should return true if there is a proposal in progress', async () => {
+      mockUseRecoveryQueue.mockReturnValue([] as any)
+
+      const { result } = renderHook(() => _useIsProposalInProgress())
+
+      expect(result.current).toBe(false)
+
+      recoveryDispatch(event, {
+        moduleAddress: faker.finance.ethereumAddress(),
+        txHash: faker.string.hexadecimal(),
+        recoveryTxHash: faker.string.hexadecimal(),
+        eventType: faker.helpers.enumValue(RecoveryEventType),
+      })
+
+      await waitFor(() => {
+        expect(result.current).toBe(true)
+      })
+    })
+  })
+  ;[RecoveryEvent.REVERTED, RecoveryEvent.PROCESSED, RecoveryEvent.FAILED].forEach((event) => {
+    it('should return false if there is not a proposal in progress', async () => {
+      const payload = {
+        moduleAddress: faker.finance.ethereumAddress(),
+        txHash: faker.string.hexadecimal(),
+        recoveryTxHash: faker.string.hexadecimal(),
+        eventType: faker.helpers.enumValue(RecoveryEventType),
```

Comment:
Same here. Shouldn't it always be RecoveryEventType.Proposal?

Reply:
Adjusted in 03398c6976df9d148a727b5e15e8ba09feb7309f.